modules/ROOT/pages/aura-graphql/api-creation.adoc

```diff
@@ -0,0 +1,161 @@
++
+Alternatively you can write your own Type Definition from first principles by following the guidance provided in xref:index.adoc[Neo4j GraphQL Library] documention
++
+If you are using GraphQL Federation then make sure to select **Enable GraphQL subgraph**.
++
+image::aura-graphql/create/type-definitions.png[]
++
+. Cross-Origin Resource Sharing (CORS) policy
++
+If a browser based application will be using this GraphQL API, add an entry in your Cross-Origin Resource Sharing (CORS) policy.
+CORS is a browser-based security mechanism that prevents web pages from accessing resources from a server on a different domain.
+Allow the domain that serves the browser application by adding it to the CORS policy.
+This includes development environments such as node.js, which serves content on your localhost or similar.
+This also holds for using web-based tooling for GraphQL APIs such as https://studio.apollographql.com/[Apollo Studio].
++
+This is not needed if a non-browser-based application is using the GraphQL API as CORS does not apply to those.
+For example, if you are trying out GraphQL operations using cURL. 
++
+[NOTE]
+====
+The domain entered in the CORS policy must be an exact match.
+For example, http://localhost is not the same as http://localhost:3000/.
+Wildcards are not supported. 
+====
++
+To add a CORS policy entry, enter the exact domain, including HTTP/S and any port number, in the **Origin box**.
+If you need multiple entries, select **Add allowed origin**.
++
+image::aura-graphql/create/cors-policy.png[]
++
+. Authentication providers
++
+All requests to the GraphQL API are authenticated and there are two options for the type of authentication: API key or JSON Web Key Set (JWKS).
+It is possible to use these in combination and have multiples of each.   
++
+After choosing the authentication provider from the dropdown list, enter a friendly name.
+This is all that is needed when using API key.
+The API key will be shown after the GraphQL API key has been created.
+If you are using JWKS, provide a friendly name and the URL from your identity provider that is used to verify JWKS tokens. 
++
+Add more more providers via **Add authentication provider**.
++
+An authentication provider can be removed by selecting the trash icon.
++
+image::aura-graphql/create/auth-providers.png[]
++
+[NOTE]
+====
+It is not recommended to use API keys with browser-based applications that are using the GraphQL API.
+This is due to the API key being visible when using browser developer tooling and very little control over GraphQL operations.
+A 3rd party identity provider with JWKS provides better security and allows for granular security rules, based on information within the JWKS token, to be defined within the Type Definitions to control GraphQL operations.
+====
++
+. Sizing
++
+Sizing a GraphQL API is based on the complexity of the type definitios, roughly their size, and the workload that is serviced.
```

Comment:
I also think we should update the wording slightly as I don't think "roughly their size" is super clear. What do you think of this?

```suggestion
Sizing a GraphQL API is based on the size and complexity of the type definitions and the workload that is serviced.
```